public/docs/dart/latest/tutorial/toh-pt6.jade

```diff
@@ -1 +1,140 @@
+
+block http-providers
+  :marked
+    Before our app can use `#{_Http}`, we have to register it as a service provider.
+
+block backend
+  :marked
+    We want to replace `BrowserClient`, the service that talks to the remote server,
+    with the in-memory web API service. 
+    Our in-memory web API service, shown below, is implemented using the
+    `http` library `MockClient` class.
+    All `http` client implementations share a common `Client` interface, so
+    we'll have our app use the `Client` type so that we can freely switch between
+    implementations.
+
+block dont-be-distracted-by-backend-subst
+  //- N/A
+
+block get-heroes-details
+  :marked
+    To get the list of heroes, we first make an asynchronous call to
+    `http.get()`. Then we use the `_extractData` helper method to decode the
+    response payload (`body`).
+
+block hero-detail-comp-extra-imports-and-vars
+  //- N/A
+
+block hero-detail-comp-updates
+  :marked
+    ### Edit in the *HeroDetailComponent*
+
+    We already have `HeroDetailComponent` for viewing details about a specific hero. 
+    Supporting edit functionality is natural extensions of the detail view, so we are able to reuse `HeroDetailComponent` with a few tweaks. 
+
+block hero-detail-comp-save-and-goback
+  //- N/A
+
+block add-new-hero-via-detail-comp
+  //- N/A
+
+block heroes-comp-directives
+  //- N/A
+
+block heroes-comp-add
+  //- N/A
+
+block review
+  //- Not showing animated gif due to differences between TS and Dart implementations.
+
+block filetree
+  .filetree
+    .file angular2-tour-of-heroes
+    .children
+      .file lib
+      .children
+        .file app_component.dart
+        .file app_component.css
+        .file dashboard_component.css
+        .file dashboard_component.html
+        .file dashboard_component.dart
+        .file hero.dart
+        .file hero_detail_component.css
+        .file hero_detail_component.html
+        .file hero_detail_component.dart
+        .file hero_service.dart
+        .file heroes_component.css
+        .file heroes_component.html
+        .file heroes_component.dart
+        .file main.dart
+        .file in_memory_data_service.dart (new)
+      .file web
+      .children
+        .file main.dart
+        .file index.html
+        .file sample.css (new)
+        .file styles.css
+      .file pubspec.yaml
+
+block file-summary
+  +makeTabs(
+    `toh-6/dart/lib/hero.dart,
+     toh-6/dart/lib/hero_detail_component.dart,
+     toh-6/dart/lib/hero_detail_component.html,
+     toh-6/dart/lib/hero_service.dart,
+     toh-6/dart/lib/heroes_component.dart,
+     toh-6/dart/web/index.html,
+     toh-6/dart/web/main.dart,
+     toh-6/dart/web/sample.css`,
+    `,,,,,,final,`,
+    `lib/hero.dart,
+     lib/hero_detail_component.dart,
+     lib/hero_detail_component.html,
+     lib/hero_service.dart,
+     lib/heroes_component.dart,
+     web/index.html,
+     web/main.dart,
+     web/sample.css`)
+
+  +makeExample('pubspec.yaml')
```

Comment:
I guess it's fine, but it seemed like a bug. A little intro would make it seem not quite so weird.

Not a showstopper.